src/software/field_tests/field_test_fixture.py

```diff
@@ -0,0 +1,629 @@
+                    always_validation_status,
+                ) = validation.run_validation_sequence_sets(
+                    world=current_world,
+                    eventually_validation_sequence_set=[[eventually_validation_set]],
+                    always_validation_sequence_set=[[]],
+                )
+
+                if not validation.contains_failure(eventually_validation_status):
+                    break
+
+            except queue.Empty as empty:
+                logger.warning("failed to obtain world")
+
+        validation.check_validation(eventually_validation_status)
+
+    def __validateMatchingRobotIds(self, worldstate: WorldState):
+        """Checks whether the robots to be used in this field test matches up with the robots we see on the field
+
+        Args:
+            worldstate (worldState Proto): The worldstate containing the robots to be used in the test
+
+        Raises:
+            Exception: On there existing a robot to be used in the test but is not on the field
+        """
+        logger.info("testing ids matching")
+
+        # validate that ids match with test setup
+        ids_present = True
+        for idx, robot_ids in enumerate(
+            [worldstate.blue_robots, worldstate.yellow_robots]
+        ):
+            team_name = "friendly" if idx == 0 else "enemy"
+            team_robots = (
+                self.friendly_robot_ids_field
+                if idx == 0
+                else self.enemy_robot_ids_field
+            )
+            for robot_id in robot_ids:
+                if robot_id not in team_robots:
+                    logger.warning(
+                        f"robot {id} from the {team_name} team present in test but not on the field"
+                    )
+                    ids_present = False
+
+        if not ids_present:
+            raise Exception("robotIds do not match")
+
+    def __setBallState(self, worldstate: WorldState):
+        """Commands a robot to move the ball to the given position on the field
+
+        Args:
+            worldstate (worldState Proto): proto that contains ball position
+
+        Raises:
+            Exception: On failing to move the ball to the correct position
+        """
+
+        logger.info("starting ball placement")
+
+        # ball placement
+        if worldstate.HasField("ball_state"):
+
+            ball_position = tbots.createPoint(worldstate.ball_state.global_position)
+
+            dribble_tactic = DribbleTactic(
+                dribble_destination=worldstate.ball_state.global_position,
+                allow_excessive_dribbling=True,
+            )
+            move_ball_tactics = AssignedTacticPlayControlParams()
+            move_ball_tactics.assigned_tactics[
+                self.friendly_robot_ids_field[0]
+            ].dribble.CopyFrom(dribble_tactic)
+
+            self.blue_full_system_proto_unix_io.send_proto(
+                AssignedTacticPlayControlParams, move_ball_tactics
+            )
```

Comment:
This function also seems to only use blue fullsystem. It seems like most of the functions in this class will need to be refactored s.t. they work with only 1 fullsystem.

Reply:
Removed